mesa/batchrunner.py

```diff
@@ -172,10 +172,15 @@ def _model_run_func(
     for step in steps:
         model_data, all_agents_data = _collect_data(model, step)
 
-        stepdata = [
-            {**{"Step": step}, **kwargs, **model_data, **agent_data}
-            for agent_data in all_agents_data
-        ]
+        # If there are agent_reporters, then create an entry for each agent
+        if all_agents_data:
+            stepdata = [
+                {**{"Step": step}, **kwargs, **model_data, **agent_data}
+                for agent_data in all_agents_data
+            ]
+        # Otherwise, just create a single entry for the step
```

Comment:
1. You didn't say that the single entry is from the model data.
2. Actually, it is confusing that the model data is put in as if it were another agent data

Reply:
OK, nvm, upon reading the code more, I saw that the model data is put in as input even in the case when there are agent_data.

Reply:
Although, I would additionally say in the comment that the entry contains only model data, without agents data.